compose-macros/src/lib.rs

```diff
@@ -101,7 +102,7 @@ macro_rules! compose_extrinsic {
                 $crate::compose_extrinsic_offline!(
                     signer,
                     call.clone(),
-                    $api.extrinsic_params($api.get_nonce().unwrap())
```

Comment:
`get_nonce` is now (potentially) async, but unfortunately the `maybe-async`-magic does not work inside of macros 😢 
So we need to get the nonce already outside and pass it in.

Reply:
This is a pity, but I never liked the `unwrap` inside the macro anyway. So I think that's an acceptable approach. But as a user-friendly macro alternative: How about offering a function called `compose_extrinisc` that does retrieve the nonce and call the macro? Just like here: 
https://github.com/scs/substrate-api-client/blob/6d09a9de7ce0fe7848d1bf2c206259bb1eceaea3/src/extrinsic/offline_extrinsic.rs#L32-L43

Reply:
The problem with the `compose_extrinisc` macro is that it takes a variable number of arguments, so it cannot as easily be wrapped into a function.

Reply:
Right, I forgot about that part. It's fine like this, but just as another idea: implementing a strictly sync version for get_nonce is not possible in this setup?

Reply:
It certainly would be nice but since we get the nonce from a node, I think it should be async. Otherwise our `async` support is quite limited when a big chunk of our requests are still sync. 
Maybe we can think of something in #318 but even there we might have to fetch the nonce from a node and then it is still `async`.
Now that I think about it a bit more.... the nonce is quite limiting for concurrency/async in general. In an async use case with a single account the likelihood of having multiple transactions with the same nonce is probably quite high. This is kind of a by design of polkadot though.

Reply:
What's the use case for an async nonce call in extrinsic creation? You have to know the nonce, otherwise you can't create the extrinsic. 

Yes, #318 would provide a different solution approach. The nonce cache would also solve your concern about the async use case. The extrinsic creation is not time-consuming if the nonce is provided locally. So no need to make it async, I believe. Only the extrinsic sending should be made available as async. If the node allows Future, the order of sending does not matter.